### PR TITLE
ios6 airplay announce message parsing bugfix  - Using Net:SDP

### PR DESCRIPTION
**Pull request by hendrikwedhorn**
I still got an SDP-Parsing error with iOS 6 beta 2 and the patch of pletoss, so i created my own patch using Net::SDP. Now it works perfectly..

### CODE REVIEW

File: shairport.pl [hunk 1]
```diff
@@ -630,20 +631,23 @@ sub conn_handle_request {
...
-            $conn->{aesiv} = $aesiv;
-            $conn->{aeskey} = $aeskey;
-            $conn->{fmtp} = $sdp{fmtp};
-            last;
-        };
+                /^ANNOUNCE$/ && do {
+                        my $sdp = Net::SDP->new($req->content);
+                        my $audio = $sdp->media_desc_of_type('audio');
+
+                        print $audio->as_string();
```
Comment (hendrikwedhorn) [inline shairport.pl:638]: These 2x print could be deletet.

**Comment by mrpippy** [comment]
This patch is also required for OS X Mountain Lion's AirPlay mirroring. Working for me with both Mountain Lion and iOS 6.

**Comment by HogeBlekker** [comment]
Indeed it works now. But, you have to try multiple times before audio comes through. Anyone else with the same problem?

**Comment by hendrikwedhorn** [comment]
No, i got instant Audio with iOS 6.0 newest Beta.

MfG Hendrik....

Am 31.07.2012 um 21:35 schrieb HogeBlekker reply@reply.github.com:

**Comment by HogeBlekker** [comment]
@hendrikw82 I can't comment on iOS 6 as I don't have it. I have to try multiple times with system-wide AirPlay audio (option click on speaker icon) on Mountain Lion 10.8.0

EDIT: recompiled using newer hairtunes.c and restarted avahi. Seems to have disappeared now. Not to worry :)

**Comment by eminence** [comment]
If this adds a new dependency (on Net::SDP), you should update INSTALL.md to reflect this

**Comment by jpoehls** [comment]
I'm hitting this error as well when streaming from iOS 6. I applied the patch locally but I get an error `Can't locate Net/SDP.pm in @inc`. I'm a Perl noob, what is the proper way to install this dependency?

**Comment by blongden** [comment]
cpan install Net::SDP worked for me - I accepted all the defaults when it was configuring cpan. You may need to run as root if you get a permissions error.

**Comment by blongden** [comment]
This patch worked perfectly for me on iOS 6 beta 4, btw.

**Comment by badboy** [comment]
+1 for this patch. Works for me on iOS 6 (final)

**Pull request merged**